File: app/src/androidTest/java/com/example/bod/kotlincoroutines/rxjavacreate/Observer.java
package com.example.bod.kotlincoroutines.rxjavacreate;

/**
 * @ClassName: Observable
 * @Description:
 * @CreateDate: 2019/10/21
 */
public interface Observer<T> {

    void onSubscribe();

    void onNext(T result);

    void onError(Throwable throwable);

    void onComplete();
}

File: app/src/main/java/com/example/bod/kotlincoroutines/create/Observer.kt
package com.example.bod.kotlincoroutines.create

/**
 *
 * @ClassName: Observer
 * @Description:
 * @CreateDate: 2019/10/21
 */
interface Observer<T> {

    fun onNext(t:T)

    fun onError()

    fun onCompleted()

}